### ISSUE TITLE
Delete app user account

### ISSUE THREAD

**Issue by sarithapillai8** [BIDMCDigitalPsychiatry/LAMP-platform#801]
Add a new menu option 'Delete account' from the settings icon as shown in the image below : 
![image](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/assets/63710347/2f5ae9d0-31af-4181-9019-334bb4234f4a)

This deletes personal information like email id, username and password which prevents the user from login again, but keeps the activity and sensor data related to that user in the database for future reference.

@carlan1 Please review

**Comment by sarithapillai8** [comment]
@carlan1 We can start working on the feature once you confirm the requirement.

**Comment by carlan1** [comment]
Yes, please work on this requirement

**Comment by sarithapillai8** [comment]
@avaidyam @carlan1 
The methods used for implementing this functionality are  LAMP.Credential.delete and LAMP.Type.setAttachment. Participant don't have permission for this API. Can you please suggest @avaidyam ?

**Comment by carlan1** [comment]
You may proceed with modifying these permissions

**Comment by sarithapillai8** [comment]
@carlan1 We have updated the changes in staging. The user will have a 'Delete account' and he/she cannot log in again. The researcher can still see the user in the listing and check the activities and sensors assigned to that user. Would you like to hide deleted users from the listing?

**Comment by sarithapillai8** [comment]
@avaidyam Nothing was made, It was possible with the participant id. Thanks

**Comment by carlan1** [comment]
External deletion request link has been added to privacy policy

**Comment by ZCOEngineer** [comment]
thank you @carlan1

**Comment by ZCOEngineer** [comment]
@carlan1 once you have submitted the missing android.permission.health.READ_HEART_RATE permission, lets us know. @sijitg is on leave till this friday, we will be able to submit the build on Friday or on coming Monday

**Comment by ZCOEngineer** [comment]
Check with @sijitg if we can submit without heart rate. ETA Friday April-05

**Comment by carlan1** [comment]
Ok, let me know if possible before making the change, Thank you

**Comment by sijitg** [comment]
@carlan1 We need to remove the code for heart rate collection, remove the permission in code and then we can submit . We can do it on Monday. Please confirm.

**Comment by carlan1** [comment]
What is the process for adding a new credential to an account that has had it previously deleted?

**Comment by sarithapillai8** [comment]
@carlan1 Do you want to list the deleted participants in the researcher section? We are currently filtering those participants. We can remove that filtering if you want to activate those users again.

**Comment by carlan1** [comment]
Yes, I think it's best to keep the user visible on the dashboard from the researcher side. Delete the login credential but keep the account, this will allow correction if an account is deleted erroneously

**Comment by sarithapillai8** [comment]
@carlan1 We have updated this in staging.

**Comment by carlan1** [comment]
The effects seem to work in staging, except when I delete an account, then go back into the account from the researcher view, then hit "delete account" for a second time, the account once again disappears from the researcher view. Could you check this?

**Comment by ZCOEngineer** [comment]
Sure @carlan1 we shall check

**Comment by sarithapillai8** [comment]
@carlan1 We have updated the staging environment. The delete account option is not needed for participants who are already deleted, so we removed that option for them. Please confirm.
